src/vocgui/views/discipline_viewset.py

```diff
@@ -31,15 +31,7 @@ def get_queryset(self):
             return None
         else:
             queryset = Discipline.objects.filter(
-                Q(released=True)
-                & Q(creator_is_admin=True)
-                & Q(
-                    id__in=[
-                        obj.id
-                        for obj in Discipline.objects.all()
-                        if obj.get_descendant_count() == 0
-                    ]
-                )
+                Q(released=True) & Q(creator_is_admin=True) & Q(lft=F("rght") - 1)
```

Comment:
```suggestion
                released=True, creator_is_admin=True, lft=F("rght") - 1
```
Just a minor detail - I guess we don't need `Q` objects when all conditions should apply.